File: PythonSolutions/004_largestPalindromeProduct.py
def isPalindrome(str):
    l = len(str)
    for i in range(0, l):
        if str[i] != str[l - 1 - i]:
            return False
    return True


def largestPalindromeProduct(digit):
    maxi = 0
    mini = 0.1
    for i in range(0, digit):
        mini = int(10 * mini)
        maxi = maxi + (9 * mini)
    num = maxi * maxi
    while num >= mini * mini:
        if isPalindrome(str(num)):
            for i in range(maxi, mini - 1, -1):
                if num % i == 0 and (num / i) <= maxi:
                    return num
        num = num - 1


print(largestPalindromeProduct(3))

File: JavaScriptSolutions/004_largestPalindromeProduct.js
const isPalindrome = str => {
    return str === str.split("").reverse().join("");
}

const largestPalindromeProduct = digit => {
    let max = 0;
    let min = 0.1;
    for (let i = 1; i <= digit; i++) {
        min *= 10;
        max = max + (9 * min);
    }
    for (let num = max * max; num > min * min; num--) {
        if(isPalindrome(num.toString())){
            for (let i = max; i >= min; i--) {
                if (num % i === 0 && (num / i) <= max) {
                    return num;
                }   
            }
        }
    }
}